server/src/main/kotlin/core/pokerhands/handUtils.kt

```diff
@@ -1,49 +1,60 @@
 package core.pokerhands
 
+import com.marcinmoskala.math.combinations
 import core.cards.Card
 import core.cards.CardRank
 
-fun makeHand(cards: Set<Card>): PokerHand {
+private inline fun <T> Set<Card>.fiveCardsMethod(f: () -> T): T {
```

Comment:
is this a decorator/?

Reply:
Not really, afaik kotlin doesn't have native support for python-like decorator. This is a regular function (I made it inline to reduce execution overhead since it is always called along with some other function). I'd recommend that you read this:
https://kotlinlang.org/docs/reference/inline-functions.html